pg_search/src/postgres/types.rs

```diff
@@ -98,6 +113,23 @@ impl TantivyValue {
                         })
                         .collect()
                 }
+                // Tantivy has a limitation that prevents JSON top-level arrays from being
+                // inserted into the index. Therefore, we need to flatten the array elements
+                // individually before converting them into Tantivy values.
+                PgBuiltInOids::JSONBOID => {
+                    let pgrx_value = pgrx::JsonB::from_datum(datum, false)
+                        .ok_or(TantivyValueError::DatumDeref)?;
+                    let json_value: Value =
+                        serde_json::from_slice(&serde_json::to_vec(&pgrx_value.0)?)?;
+                    Ok(Self::json_value_to_tantivy_value(json_value))
+                }
+                PgBuiltInOids::JSONOID => {
+                    let pgrx_value = pgrx::JsonB::from_datum(datum, false)
```

Comment:
Note that this should have used `pgrx::Json::from_datum()`.  I've PR'd a fix (with test) in PR #1654.